hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/execution/bulkinsert/TestBulkInsertInternalPartitionerForRows.java

```diff
@@ -116,36 +115,24 @@ public void testBulkInsertInternalPartitioner(BulkInsertSortMode sortMode,
 
     testBulkInsertInternalPartitioner(
         BulkInsertInternalPartitionerWithRowsFactory.get(config, isTablePartitioned, enforceNumOutputPartitions),
-        records1,
+        records,
         enforceNumOutputPartitions,
         isGloballySorted,
         isLocallySorted,
-        generateExpectedPartitionNumRecords(records1),
-        Option.empty(),
-        populateMetaFields);
-    testBulkInsertInternalPartitioner(
-        BulkInsertInternalPartitionerWithRowsFactory.get(config, isTablePartitioned, enforceNumOutputPartitions),
-        records2,
-        enforceNumOutputPartitions,
-        isGloballySorted,
-        isLocallySorted,
-        generateExpectedPartitionNumRecords(records2),
+        generateExpectedPartitionNumRecords(records),
         Option.empty(),
         populateMetaFields);
   }
 
   @Test
   public void testCustomColumnSortPartitionerWithRows() {
-    Dataset<Row> records1 = generateTestRecords();
-    Dataset<Row> records2 = generateTestRecords();
-    String sortColumnString = records1.columns()[5];
+    Dataset<Row> records = generateTestRecords();
+    String sortColumnString = records.columns()[5];
     String[] sortColumns = sortColumnString.split(",");
     Comparator<Row> comparator = getCustomColumnComparator(sortColumns);
 
     testBulkInsertInternalPartitioner(new RowCustomColumnsSortPartitioner(sortColumns),
-        records1, true, false, true, generateExpectedPartitionNumRecords(records1), Option.of(comparator), true);
-    testBulkInsertInternalPartitioner(new RowCustomColumnsSortPartitioner(sortColumns),
-        records2, true, false, true, generateExpectedPartitionNumRecords(records2), Option.of(comparator), true);
+        records, true, true, true, generateExpectedPartitionNumRecords(records), Option.of(comparator), true);
```

Comment:
the existing test treated the partitioner as non-global and hence failed the test scenario under spark 3.2